apps/system/js/notifications.js

```diff
@@ -408,9 +408,13 @@ var NotificationScreen = {
     // timesout, so we skip updating it here, by passing a skip bool
     this.addUnreadNotification(detail.id, true);
 
+    var cIndex = this.availableNotification.indexOf(detail.appName);
+    if (cIndex === -1) {
+    this.availableNotification.push(detail.appName);
     var notificationNode = document.createElement('div');
     notificationNode.classList.add('notification');
     notificationNode.setAttribute('role', 'link');
+    notificationNode.setAttribute('id', detail.appName);
```

Comment:
appName is not a good identifier. It changes when locale changes, and there can be multiple apps with the same name.  Use the app's manifestURL as a unique string instead.